api/rest/restapi_test.go

```diff
@@ -1039,6 +1040,38 @@ func TestAPIRecoverAllEndpoint(t *testing.T) {
 	testBothEndpoints(t, tf)
 }
 
+func TestAPILogging(t *testing.T) {
+	ctx := context.Background()
+	cfg := &Config{}
+	cfg.Default()
+
+	cfg.HTTPLogFile = "/tmp/http.log"
+
+	rest := testAPIwithConfig(t, cfg, "log_enabled")
+	defer rest.Shutdown(ctx)
+	defer os.Remove(cfg.HTTPLogFile)
+
+	data, err := ioutil.ReadFile(cfg.HTTPLogFile)
```

Comment:
check out https://golang.org/pkg/os/#Stat . But I'm not sure we have to test that an empty file exists right away. Better to test would be to make sure than an existing file is not replaced on a subsequent start.